server/src/main/java/org/opensearch/index/mapper/KnnAlgorithmContext.java

```diff
@@ -0,0 +1,176 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.index.mapper;
+
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Abstracts KNN Algorithm segment of dense_vector field type
+ */
+public class KnnAlgorithmContext implements ToXContentFragment, Writeable {
+
+    private static final String PARAMETERS = "parameters";
+    private static final String NAME = "name";
+
+    private final Method method;
+    private final Map<String, Object> parameters;
+
+    private static final int MAX_NUMBER_OF_ALGORITHM_PARAMETERS = 50;
```

Comment:
nit: should we move all constants to beginning of class?

Reply:
ack